src/components/atoms/InfiniteScroll/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { ReactNode, useEffect, useRef } from "react";
+
+import { styled } from "@reearth/theme";
+
+import useHooks from "./hooks";
+
+type Props = {
+  className?: string;
+  loading?: boolean;
+  hasMoreItems?: boolean;
+  children?: ReactNode;
+  onGetMoreItems?: () => void;
+};
+
+const InfiniteScrollWrapper: React.FC<Props> = ({
+  className,
+  loading,
+  hasMoreItems,
+  onGetMoreItems,
+  children,
+}) => {
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const { handleAutoFillPage, handleScrollToBottom } = useHooks(onGetMoreItems);
+
+  useEffect(() => {
+    if (wrapperRef.current && !loading && hasMoreItems) handleAutoFillPage(wrapperRef);
+  }, [handleAutoFillPage, hasMoreItems, loading, onGetMoreItems]);
```

Comment:
This component doesn't have too many hooks, so you don't really need the hooks.ts file. But if you want to keep the file, please bring the ref and useEffect into the hooks file as well to keep things consistent (unless there is a reason you separated them, in which case please let me know).